addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
Well, i think we should actually do this, but we should doit for the right reasons. It exists, to correctly enable batching of XHR. as ed does it https://github.com/emberjs/data/blob/master/addon/adapters/rest.js#L857-L859